Fix 17.x site build

gregorydlogan opened the pull request:
The site plugin requires install to have run, lest it be unable to find
artifacts.  This makes sense since install is what actually puts them
into your repo.  The catch here is that install + site in the same build
hits issues with checkerqual, which conflicts with site.  The solution
is a two pass approach, which avoids the conflict.

We deliberately disabled the first build-only test pass since we don't care
about the results of the tests *and* we still run the tests in the site
pass.  If we don't build the tests during the site pass we don't get the
code coverage metrics that we're after with the site build in the first
place.

This should go into 17 since this has broken the build in 17, and having
coverage data for 17 would be useful.

### Your pull request should…

* [x] have a concise title
* [x] ~[close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists~
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] ~include migration scripts and documentation, if appropriate~
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [x] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

gregorydlogan committed:
Fix 17.x site build

The site plugin requires install to have run, lest it be unable to find
artifacts.  This makes sense since install is what actually puts them
into your repo.  The catch here is that install + site in the same build
hits issues with checkerqual, which conflicts with site.  The solution
is a two pass approach, which avoids the conflict.

We deliberately disabled the first build-only pass since we don't care
about the results of the test *and* we still run the tests in the site
pass.  If we don't build the tests during the site pass we don't get the
code coverage metrics that we're after with the site build in the first
place.

gregorydlogan commented:
The checkerqual stuff I was talking about was fixed in https://github.com/opencast/opencast/commit/e9e1f29862e8b33bb2f1bf9105902d4b5b1e0d62

lkiesow commented on .github/workflows/test-opencast-site.yml:
```diff
@@ -67,17 +67,27 @@ jobs:
+        run: |
+          ./mvnw install -DskipTests -T1C -Pnone \
+            --batch-mode \
+            -Dsurefire.rerunFailingTestsCount=2 \
```
We don't actually need this since we don't run any tests, do we?
We could skip the style checker, though.
```suggestion
            -Dcheckstyle.skip=true \
```

gregorydlogan replied:
Yep, great catch

gregorydlogan committed:
Incorporating review suggestions

Don't need to worry about failing tests when tests aren't running.  And we can check styles elsewhere/later.

Co-authored-by: Lars Kiesow <lkiesow@uos.de>
